# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build evas-build"

PKG_CONFIG_PATH=/fb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.67; am-1.11; lt-2.2.6b)
libtoolize -c -f; aclocal -I m4; autoheader; autoconf; automake -a -c
PKG_CONFIG_PATH=/fb/lib/pkgconfig LDFLAGS=-Wl,-rpath,/fb/lib ./configure --prefix=/fb

# build
make

# install
install -d $DESTDIR/fb/bin
install src/bin/expedite $DESTDIR/fb/bin
install src/bin/expedite-cmp $DESTDIR/fb/bin
install -d $DESTDIR/fb/share/expedite/data
install -m 644 data/3d.png $DESTDIR/fb/share/expedite/data
install -m 644 data/bar.png $DESTDIR/fb/share/expedite/data
install -m 644 data/blend.png $DESTDIR/fb/share/expedite/data
install -m 644 data/bug.png $DESTDIR/fb/share/expedite/data
install -m 644 data/bulb.png $DESTDIR/fb/share/expedite/data
install -m 644 data/camera.png $DESTDIR/fb/share/expedite/data
install -m 644 data/colorbox.png $DESTDIR/fb/share/expedite/data
install -m 644 data/cube1.png $DESTDIR/fb/share/expedite/data
install -m 644 data/cube2.png $DESTDIR/fb/share/expedite/data
install -m 644 data/cube3.png $DESTDIR/fb/share/expedite/data
install -m 644 data/cube4.png $DESTDIR/fb/share/expedite/data
install -m 644 data/cube5.png $DESTDIR/fb/share/expedite/data
install -m 644 data/cube6.png $DESTDIR/fb/share/expedite/data
install -m 644 data/data.png $DESTDIR/fb/share/expedite/data
install -m 644 data/e.png $DESTDIR/fb/share/expedite/data
install -m 644 data/e-logo-2.png $DESTDIR/fb/share/expedite/data
install -m 644 data/e-logo-mask.png $DESTDIR/fb/share/expedite/data
install -m 644 data/e-logo.png $DESTDIR/fb/share/expedite/data
install -m 644 data/error.png $DESTDIR/fb/share/expedite/data
install -m 644 data/exit.png $DESTDIR/fb/share/expedite/data
install -m 644 data/flower.png $DESTDIR/fb/share/expedite/data
install -m 644 data/fonts.alias $DESTDIR/fb/share/expedite/data
install -m 644 data/fonts.dir $DESTDIR/fb/share/expedite/data
install -m 644 data/frame.png $DESTDIR/fb/share/expedite/data
install -m 644 data/house.png $DESTDIR/fb/share/expedite/data
install -m 644 data/icon_sel.png $DESTDIR/fb/share/expedite/data
install -m 644 data/im1.png $DESTDIR/fb/share/expedite/data
install -m 644 data/im2.png $DESTDIR/fb/share/expedite/data
install -m 644 data/image.png $DESTDIR/fb/share/expedite/data
install -m 644 data/Kochi.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/logo.png $DESTDIR/fb/share/expedite/data
install -m 644 data/map.png $DESTDIR/fb/share/expedite/data
install -m 644 data/mushroom.png $DESTDIR/fb/share/expedite/data
install -m 644 data/pan.png $DESTDIR/fb/share/expedite/data
install -m 644 data/pulse.png $DESTDIR/fb/share/expedite/data
install -m 644 data/rect.png $DESTDIR/fb/share/expedite/data
install -m 644 data/text.png $DESTDIR/fb/share/expedite/data
install -m 644 data/text_sel.png $DESTDIR/fb/share/expedite/data
install -m 644 data/texture.png $DESTDIR/fb/share/expedite/data
install -m 644 data/tp.png $DESTDIR/fb/share/expedite/data
install -m 644 data/tp.yuv $DESTDIR/fb/share/expedite/data
install -m 644 data/typewriter.png $DESTDIR/fb/share/expedite/data
install -m 644 data/Vera.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/VeraBI.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/VeraBd.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/VeraIt.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/VeraMoBI.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/VeraMoBd.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/VeraMoIt.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/VeraMono.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/VeraSe.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/VeraSeBd.ttf $DESTDIR/fb/share/expedite/data
install -m 644 data/warning.png $DESTDIR/fb/share/expedite/data
install -m 644 data/watch.png $DESTDIR/fb/share/expedite/data
install -m 644 data/widgets.png $DESTDIR/fb/share/expedite/data

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/expedite-build.pc << EOF
Name: expedite
Version: 1.2.0
Description: A collection of Evas demos and test programs
Requires: $REQUIRES

exec=\\
/fb/bin/expedite \\
/fb/bin/expedite-cmp \\
/fb/share/expedite/data/3d.png \\
/fb/share/expedite/data/bar.png \\
/fb/share/expedite/data/blend.png \\
/fb/share/expedite/data/bug.png \\
/fb/share/expedite/data/bulb.png \\
/fb/share/expedite/data/camera.png \\
/fb/share/expedite/data/colorbox.png \\
/fb/share/expedite/data/cube1.png \\
/fb/share/expedite/data/cube2.png \\
/fb/share/expedite/data/cube3.png \\
/fb/share/expedite/data/cube4.png \\
/fb/share/expedite/data/cube5.png \\
/fb/share/expedite/data/cube6.png \\
/fb/share/expedite/data/data.png \\
/fb/share/expedite/data/e.png \\
/fb/share/expedite/data/e-logo-2.png \\
/fb/share/expedite/data/e-logo-mask.png \\
/fb/share/expedite/data/e-logo.png \\
/fb/share/expedite/data/error.png \\
/fb/share/expedite/data/exit.png \\
/fb/share/expedite/data/flower.png \\
/fb/share/expedite/data/fonts.alias \\
/fb/share/expedite/data/fonts.dir \\
/fb/share/expedite/data/frame.png \\
/fb/share/expedite/data/house.png \\
/fb/share/expedite/data/icon_sel.png \\
/fb/share/expedite/data/im1.png \\
/fb/share/expedite/data/im2.png \\
/fb/share/expedite/data/image.png \\
/fb/share/expedite/data/Kochi.ttf \\
/fb/share/expedite/data/logo.png \\
/fb/share/expedite/data/map.png \\
/fb/share/expedite/data/mushroom.png \\
/fb/share/expedite/data/pan.png \\
/fb/share/expedite/data/pulse.png \\
/fb/share/expedite/data/rect.png \\
/fb/share/expedite/data/text.png \\
/fb/share/expedite/data/text_sel.png \\
/fb/share/expedite/data/texture.png \\
/fb/share/expedite/data/tp.png \\
/fb/share/expedite/data/tp.yuv \\
/fb/share/expedite/data/typewriter.png \\
/fb/share/expedite/data/Vera.ttf \\
/fb/share/expedite/data/VeraBI.ttf \\
/fb/share/expedite/data/VeraBd.ttf \\
/fb/share/expedite/data/VeraIt.ttf \\
/fb/share/expedite/data/VeraMoBI.ttf \\
/fb/share/expedite/data/VeraMoBd.ttf \\
/fb/share/expedite/data/VeraMoIt.ttf \\
/fb/share/expedite/data/VeraMono.ttf \\
/fb/share/expedite/data/VeraSe.ttf \\
/fb/share/expedite/data/VeraSeBd.ttf \\
/fb/share/expedite/data/warning.png \\
/fb/share/expedite/data/watch.png \\
/fb/share/expedite/data/widgets.png
EOF

exit

# patch
--- expedite-1.2.0.orig/configure.ac
+++ expedite-1.2.0/configure.ac
@@ -11,6 +11,7 @@
 AC_CANONICAL_HOST
 
 AM_INIT_AUTOMAKE([1.6 dist-bzip2])
+AM_MAINTAINER_MODE
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
 PKG_PROG_PKG_CONFIG
@@ -27,6 +28,11 @@
    [want_evas_simple_x11=$enableval]
 )
 
+AC_ARG_ENABLE(sdl,
+   [AC_HELP_STRING([--enable-sdl], [enable SDL engine])],
+   [want_sdl=$enableval], [have_sdl="no"]
+)
+
 # Software X11
 have_software_xlib="no"
 have_software_xcb="no"
@@ -117,7 +123,7 @@
 AC_PROG_CXX
 AC_PROG_CC
 
-LT_PROG_RC
+LT_INIT
 
 
 
@@ -138,7 +144,7 @@
 
 # Evas
 
-PKG_CHECK_MODULES([EVAS], evas >= 1.2.0 eet >= 1.6.0)
+PKG_CHECK_MODULES([EVAS], evas >= 1.2.0)
 
 # Xlib
 
@@ -172,7 +178,9 @@
 
 # SDL
 
-PKG_CHECK_MODULES([SDL], [sdl], [have_sdl="yes"], [have_sdl="no"])
+if test "x$want_sdl" = "xyes"; then
+   PKG_CHECK_MODULES([SDL], [sdl], [have_sdl="yes"], [have_sdl="no"])
+fi
 AM_CONDITIONAL(BUILD_SDL, test "x$have_sdl" = "xyes")
 if test "x$have_sdl" = "xyes"; then
    AC_DEFINE(BUILD_SDL, 1, [build sdl support])
@@ -266,7 +274,7 @@
 echo "    Software GDI.................: ${have_software_gdi}"
 echo "    Software DirectDraw..........: ${have_software_ddraw}"
 echo "    Direct3D.....................: ${have_direct3d}"
-echo "    Software SDL.................: ${have_software_sdl}"
+echo "    Software SDL.................: ${have_sdl}"
 echo "    Open GL SDL..................: ${have_opengl_sdl}"
 echo "    Open GL Cocoa................: ${have_opengl_cocoa}"
 echo "    FB...........................: ${have_fb}"
--- expedite-1.2.0.orig/src/bin/Makefile.am
+++ expedite-1.2.0/src/bin/Makefile.am
@@ -17,6 +17,8 @@
 bin_PROGRAMS = expedite
 bin_SCRIPTS = expedite-cmp
 
+noinst_LTLIBRARIES =
+
 expedite_SOURCES = \
 main.c main.h \
 ui.c ui.h \
@@ -178,13 +180,17 @@
 endif
 
 if BUILD_SOFTWARE_DDRAW
-expedite_SOURCES += \
+noinst_LTLIBRARIES += libddraw.la
+libddraw_la_SOURCES = \
 engine_software_ddraw.cpp engine_software_ddraw.h
+expedite_LDADD += libddraw.la
 endif
 
 if BUILD_DIRECT3D
-expedite_SOURCES += \
+noinst_LTLIBRARIES += libdirect3d.la
+libdirect3d_la_SOURCES = \
 engine_direct3d.cpp engine_direct3d.h
+expedite_LDADD += libdirect3d.la
 endif
 
 if BUILD_SDL
@@ -223,8 +229,10 @@
 endif
 
 if BUILD_SOFTWARE_16_DDRAW
-expedite_SOURCES += \
+noinst_LTLIBRARIES += libddraw16.la
+libddraw16_la_SOURCES = \
 engine_software_16_ddraw.cpp engine_software_16_ddraw.h
+expedite_LDADD += libddraw16.la
 endif
 
 if BUILD_SOFTWARE_16_WINCE
--- expedite-1.2.0.orig/src/bin/engine_fb.c
+++ expedite-1.2.0/src/bin/engine_fb.c
@@ -1,6 +1,12 @@
 #include "main.h"
 
 #include <Evas_Engine_FB.h>
+#include <dirent.h>
+#include <fcntl.h>
+#include <linux/input.h>
+
+static int fd = -1;
+static DIR *dir = NULL;
 
 Eina_Bool
 engine_fb_args(const char *engine, int width, int height)
@@ -16,6 +22,37 @@
 	return EINA_FALSE;
      }
 
+   dir = opendir("/dev/input");
+   if (dir)
+     {
+        char path[32];
+        struct dirent *entry;
+        unsigned char bits[(KEY_CNT - 1) / 8 + 1];
+
+        while ((entry = readdir(dir)))
+          {
+             if (entry->d_type == DT_CHR)
+               {
+                  sprintf(path, "/dev/input/%s", entry->d_name);
+                  fd = open(path, O_RDONLY | O_NONBLOCK);
+                  if (fd == -1)
+                    continue;
+
+                  if (ioctl(fd, EVIOCGBIT(EV_KEY, sizeof(bits)), bits) == -1)
+                    {
+                       close(fd);
+                       continue;
+                    }
+
+                  if (bits[KEY_ENTER / 8] & (1 << (KEY_ENTER % 8)))
+                    break;
+
+                  close(fd);
+                  fd = -1;
+               }
+          }
+     }
+
    einfo->info.virtual_terminal = 0;
    einfo->info.device_number = 0;
    einfo->info.device_number = 0;
@@ -33,11 +70,54 @@
 void
 engine_fb_loop(void)
 {
-   return;
+   const char *kstr;
+   struct input_event iev;
+
+   if (fd == -1)
+     return;
+
+   if (read(fd, &iev, sizeof(struct input_event)) > 0 && iev.type == EV_KEY)
+     {
+        switch (iev.code)
+          {
+           case KEY_LEFT:
+             kstr = "Left";
+             break;
+
+           case KEY_RIGHT:
+             kstr = "Right";
+             break;
+
+           case KEY_ENTER:
+           case KEY_KPENTER:
+             kstr = "Return";
+             break;
+
+           case KEY_ESC:
+             kstr = "Escape";
+             break;
+
+           default:
+             kstr = NULL;
+             break;
+          }
+
+        if (kstr)
+          {
+             if (iev.value)
+               evas_event_feed_key_down(evas, kstr, kstr, NULL, NULL, 0, NULL);
+             else
+               evas_event_feed_key_up(evas, kstr, kstr, NULL, NULL, 0, NULL);
+          }
+     }
 }
 
 void
 engine_fb_shutdown(void)
 {
-   return;
+   if (fd != -1)
+     close(fd);
+
+   if (dir)
+     closedir(dir);
 }
--- expedite-1.2.0.orig/src/bin/engine_software_sdl.c
+++ expedite-1.2.0/src/bin/engine_software_sdl.c
@@ -1,7 +1,7 @@
 #include <assert.h>
 #include "main.h"
 
-#include <SDL/SDL.h>
+#include <SDL.h>
 #include <Evas_Engine_Buffer.h>
 
 static void *
--- expedite-1.2.0.orig/src/bin/expedite-cmp
+++ expedite-1.2.0/src/bin/expedite-cmp
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/bin/env python
 
 import sys
 import os
--- expedite-1.2.0.orig/src/bin/main.c
+++ expedite-1.2.0/src/bin/main.c
@@ -1,8 +1,8 @@
 #include "main.h"
 
 Evas *evas = NULL;
-const char *profile = "n800";
-int win_w = 720, win_h = 420;
+const char *profile = "svga";
+int win_w = 800, win_h = 600;
 const char *engine = NULL;
 int loops = LOOPS;
 int fullscreen = 0;

# source
https://raw.githubusercontent.com/caramelli/efl-legacy/master/expedite-1.2.0.tar.gz
